commit 5a49ef9832e599d2fba7fae80049067cd7c5950f
Author: Parav Pandit <parav@mellanox.com>
Date:   Mon Nov 19 09:58:24 2018 +0200

    RDMA/core: Add GIDs while changing MAC addr only for registered ndev
    
    [ Upstream commit d52ef88a9f4be523425730da3239cf87bee936da ]
    
    Currently when MAC address is changed, regardless of the netdev reg_state,
    GID entries are removed and added to reflect the new MAC address and new
    default GID entries.
    
    When a bonding device is used and the underlying PCI device is removed
    several netdevice events are generated. Two events of the interest are
    CHANGEADDR and UNREGISTER event on lower(slave) netdevice of the bond
    netdevice.
    
    Sometimes CHANGEADDR event is generated when netdev state is
    UNREGISTERING (after UNREGISTER event is generated). In this scenario, GID
    entries for default GIDs are added and never deleted because GID entries
    are deleted only when netdev state is < UNREGISTERED.
    
    This leads to non zero reference count on the netdevice. Due to this, PCI
    device unbind operation is getting stuck.
    
    To avoid it, when changing mac address, add GID entries only if netdev is
    in REGISTERED state.
    
    Fixes: 03db3a2d81e6 ("IB/core: Add RoCE GID table management")
    Signed-off-by: Parav Pandit <parav@mellanox.com>
    Reviewed-by: Mark Bloch <markb@mellanox.com>
    Signed-off-by: Leon Romanovsky <leonro@mellanox.com>
    Signed-off-by: Jason Gunthorpe <jgg@mellanox.com>
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/infiniband/core/roce_gid_mgmt.c b/drivers/infiniband/core/roce_gid_mgmt.c
index ee366199b169..25d43c8f1c2a 100644
--- a/drivers/infiniband/core/roce_gid_mgmt.c
+++ b/drivers/infiniband/core/roce_gid_mgmt.c
@@ -767,8 +767,10 @@ static int netdevice_event(struct notifier_block *this, unsigned long event,
 
 	case NETDEV_CHANGEADDR:
 		cmds[0] = netdev_del_cmd;
-		cmds[1] = add_default_gid_cmd;
-		cmds[2] = add_cmd;
+		if (ndev->reg_state == NETREG_REGISTERED) {
+			cmds[1] = add_default_gid_cmd;
+			cmds[2] = add_cmd;
+		}
 		break;
 
 	case NETDEV_CHANGEUPPER:
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1071_5a49ef9832e599d2fba7fae80049067cd7c5950f.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1071_5a49ef9832e599d2fba7fae80049067cd7c5950f.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
